File: netlify/functions/dualUpload.py
# .netlify/functions/dualUpload.py
import json
import tempfile
import os
import pyreadstat
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/.netlify/functions/dualUpload', methods=['POST'])
def upload():
    if 'file' not in request.files:
        return jsonify({"error": "No file uploaded"}), 400

    file = request.files['file']
    filename = file.filename.lower()

    try:
        with tempfile.TemporaryDirectory() as tmpdir:
            filepath = os.path.join(tmpdir, file.filename)
            file.save(filepath)

            if filename.endswith('.sav'):
                df, meta = pyreadstat.read_sav(filepath)
            elif filename.endswith('.csv'):
                import pandas as pd
                df = pd.read_csv(filepath)
            elif filename.endswith('.xlsx'):
                import pandas as pd
                df = pd.read_excel(filepath)
            else:
                return jsonify({"error": "Unsupported file type"}), 400

            summary = {
                "rows": len(df),
                "columns": len(df.columns),
                "columns_info": []
            }

            for col in df.columns:
                col_info = {
                    "name": col,
                    "dtype": str(df[col].dtype),
                    "missing": int(df[col].isnull().sum()),
                    "unique": int(df[col].nunique())
                }
                summary["columns_info"].append(col_info)

            return jsonify(summary)

    except Exception as e:
        return jsonify({"error": str(e)}), 500

# Needed for Netlify Python Function compatibility
def handler(event, context):
    from werkzeug.datastructures import FileStorage
    from werkzeug.wrappers import Request
    
    request = Request(event['rawBody'])
    return upload()

File: netlify/functions/dualUpload.js
// File: netlify/functions/dualUpload.js

const fs = require('fs');
const path = require('path');
const os = require('os');
const multiparty = require('multiparty');
const py = require('child_process');

exports.handler = async (event, context) => {
  if (event.httpMethod !== 'POST') {
    return {
      statusCode: 405,
      body: 'Method Not Allowed',
    };
  }

  const form = new multiparty.Form();

  return new Promise((resolve, reject) => {
    form.parse(event, async (err, fields, files) => {
      if (err) {
        return resolve({ statusCode: 500, body: 'Upload error: ' + err.message });
      }

      const uploaded = files.file?.[0];
      if (!uploaded) {
        return resolve({ statusCode: 400, body: 'No file uploaded.' });
      }

      const ext = path.extname(uploaded.originalFilename).toLowerCase();
      const tmpPath = uploaded.path;
      let pythonScript;

      if (ext === '.sav') {
        pythonScript = 'analyze_sav.py';
      } else if (ext === '.xlsx' || ext === '.xls') {
        pythonScript = 'analyze_excel.py';
      } else if (ext === '.csv') {
        pythonScript = 'analyze_csv.py';
      } else {
        return resolve({
          statusCode: 400,
          body: 'Unsupported file type: ' + ext,
        });
      }

      try {
        const pyProcess = py.spawnSync('python3', [path.join(__dirname, pythonScript), tmpPath]);

        if (pyProcess.error) {
          return resolve({ statusCode: 500, body: 'Python error: ' + pyProcess.error.message });
        }

        return resolve({
          statusCode: 200,
          body: pyProcess.stdout.toString(),
        });
      } catch (error) {
        return resolve({
          statusCode: 500,
          body: 'Error running Python script: ' + error.message,
        });
      }
    });
  });
};
